test/inductor/test_cpu_repro.py

```diff
@@ -2892,6 +2892,18 @@ def fn(out1, out2, input, other):
             with torch.no_grad():
                 self.common(fn, (o1, o2, x, y))
 
+    @torch._dynamo.config.patch(dynamic_shapes=True)
+    @torch._dynamo.config.patch(assume_static_by_default=False)
+    def test_symbolic_shape(self):
+        def fn(x, y):
+            return y + torch.ones(x).sum()
+
+        with torch.no_grad():
+            metrics.reset()
+            y = torch.randn(100)
+            self.common(fn, (100, y))
+            assert metrics.generated_cpp_vec_kernel_count == 2
```

Comment:
How come we are generating two kernels here? Using the bounds on the variables we should even be able to go from `ones(x).sum()` to `x`.

Reply:
There is a kernel for reduction and the other one for add. It is with dynamic shape and the size of reduction and the size for add could be different, hence no fusion happens.

Reply:
I thought that since both of the args are equal to `100` they would be allocated the same symbol (we call that duckshaping). But perhaps we don't do that under `dynamic=True`? cc  @ezyang just in case.

Reply:
There's a separate issue here. I filled https://github.com/pytorch/pytorch/issues/118521